lib/appmap/util.rb

```diff
@@ -72,6 +80,19 @@ def sanitize_event(event, &block)
         event
       end
 
+      def select_headers(env)
+        # Rack prepends HTTP_ to all client-sent headers.
+        matching_headers = env
+          .select { |k,v| k.start_with? 'HTTP_'}
+          .reject { |k,v| v.blank? }
+          .each_with_object({}) do |kv, memo|
+            key = kv[0].sub(/^HTTP_/, '').split('_').map(&:capitalize).join('-')
+            value = kv[1]
+            memo[key] = value
+          end
+        matching_headers.blank? ? nil : matching_headers
+      end
```

Comment:
Doesn't this just reimplement `request.headers` now (with the exception of niling if empty)?